scripts/new_upstream_snapshot.py

```diff
@@ -133,7 +133,7 @@ def replace(
         return VersionInfo(
             major=major or self.major,
             minor=minor or self.minor,
-            hotfix=hotfix if hotfix is not None else self.hotfix,
+            hotfix=hotfix if hotfix or any((major, minor)) else self.hotfix,
```

Comment:
I guess we would probably never have a major without a minor, but if we did we'd need minor=minor if major else self.minor, I think?

Not asking for changes, trying to understand the code in different scenarios.

Reply:
Yes, in our current call-sites we'd never have a major without a minor. Also logistically, if we ever bump a major, we aren't going to want a major without a minor as far as release policy. I can imaging 25.3 -> 26.0 potentially, our minor should always be present and we won't be incrementing/jumping a major without also setting a minor anywhere.